lld/ELF/ScriptParser.cpp

```diff
@@ -547,10 +549,15 @@ SmallVector<SectionCommand *, 0> ScriptParser::readOverlay() {
     // starting from the base load address specified.
     OutputDesc *osd = readOverlaySectionDescription();
     osd->osec.addrExpr = addrExpr;
-    if (prev)
+    if (prev) {
       osd->osec.lmaExpr = [=] { return prev->getLMA() + prev->size; };
-    else
+    } else {
       osd->osec.lmaExpr = lmaExpr;
+      // Ensure that the first section, even if empty, is not discarded.
+      // Subsequent sections share its address.
+      osd->osec.usedInExpression = true;
+      addrExpr = [=]() -> ExprValue { return {&osd->osec, false, 0, ""}; };
```

Comment:
I guess addrExpr is for the case where the original addrExpr is DOT. May be worth updating the comment to.
```
// Use first section address for subsequent sections as initial addrExpr can be DOT. 
```

Reply:
Thanks for the suggestion. Adopted!

Reply:
Thanks for the suggestion. I reworded it a bit.
```
+  // When AT is omitted, LMA should equal VMA. script->getDot() when evaluating
+  // lmaExpr will ensure this, even if the start address is specified.
```

Setting `lmaExpr` to `script->getDot()` ensures that LMA propagation in `LinkerScript::assignOffsets` is disabled and `state->lmaOffset = sec->lmaExpr().getValue() - dot;` evaluates to 0.